### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by IvanGuculyak**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    results = dict()
+
+    def inner(*args: Any) -> Any:
+
+        if args in results.keys():
```
Comment (olena-hul) [inline app/main.py:9]: `.keys()` is redundant

**Review by olena-hul — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    results = dict()
```
Comment (Vasyl-Smutok) [inline app/main.py:5]: try to choose a better name than `result`
`cache_dict` or something like that

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,16 @@
...
+
+    def inner(*args: Any) -> Any:
```
Comment (Vasyl-Smutok) [inline app/main.py:7]: I would recommend using the name `wrapper`

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    results = dict()
+
```
Comment (Vasyl-Smutok) [inline app/main.py:6]: is also a good habit in decorators to use
`@wraps(func)`

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Review by vsmutok — APPROVED** [review]
Good Job 👍

**Pull request closed without merging**